### PR TITLE
Fail the build if multiple container-image extensions are available

### LINKED ISSUE #8584: A project with both docker and jib extension  - no warning about this situation and image gets build twice under the same name
A project with both docker and jib extension  - no warning about this situation and the image gets build twice under the same name.

Expectation:
 - warning about the situation
 - optionally: only one build, but who should win?

I did some experiments with docker build and wanted to see the differences in jib image so I added the jib extension. But the docker image produced by the new run had the same image size as the previous one. So that made me a bit worried before I realized two builds happened during the mvn command execution.

Reproducer:
```bash
mvn io.quarkus:quarkus-maven-plugin:1.3.2.Final:create \
    -DprojectGroupId=org.acme \
    -DprojectArtifactId=getting-started \
    -DclassName="org.acme.getting.started.GreetingResource" \
    -Dpath="/hello" -Dextensions="docker,jib"
cd getting-started
mvn clean package -Dquarkus.container-image.build=true
```

### PR DESCRIPTION
**Pull request by geoand**
Fixes: #8584

### CODE REVIEW

**Comment by geoand** [comment]
Opened as draft to give my own GH Actions the chance to run CI.

**Comment by Ladicek** [comment]
Looks good on the first sight. Just wondering if `container-image-s2i` also collides with `container-image-{docker,jib}`?

**Comment by geoand** [comment]
I think it should - but that is where I need input from @iocanel.

File: integration-tests/container-image/pom.xml [hunk 1]
```diff
@@ -7,91 +7,13 @@
...
-      </properties>
-    </profile>
-  </profiles>
+  <artifactId>quarkus-integration-test-container-image-parent</artifactId>
+  <name>Quarkus - Integration Tests - Container Image - Parent</name>
+  <packaging>pom</packaging>
+
+  <modules>
+    <module>invoker</module>
+    <module>standard</module>
```
Comment (rsvoboda) [inline integration-tests/container-image/pom.xml:17]: I don't like `standard` name. Can be more descriptive name picked?

Comment (geoand) [reply integration-tests/container-image/pom.xml:17]: Do you have a suggestion? I suck horribly at naming.

FWIW we already use that name in kubernetes tests.

**Comment by rsvoboda** [comment]
Current exception is `java.lang.IllegalStateException: At most one container-image extension can be present.`

Could the error message be more specific and say something like:
`quarkus-container-image-jib and quarkus-container-image-docker were detected, at most one container-image extension can be present.` ?

```
[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:999-SNAPSHOT:build (default) on project getting-started: Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
[ERROR] 	[error]: Build step io.quarkus.container.image.deployment.ContainerProcessor#publishImageInfo threw an exception: java.lang.IllegalStateException: At most one container-image extension can be present.
[ERROR] 	at io.quarkus.container.image.deployment.ContainerProcessor.publishImageInfo(ContainerProcessor.java:15)
```

File: integration-tests/container-image/pom.xml (hunk 1 above)
Comment (rsvoboda) [reply integration-tests/container-image/pom.xml:17]: so there is maven-invoker way and standard Quarkus way of working with container-image

what about:
 - invoker => maven-invoker-way
 - standard => quarkus-standard-way
?

**Comment by geoand** [comment]
Sure, we can do that

File: integration-tests/container-image/pom.xml (hunk 1 above)
Comment (geoand) [reply integration-tests/container-image/pom.xml:17]: I don't particularly like `way` in there TBH. @gsmet any good suggestions here?

**Comment by geoand** [comment]
Taking PR out of draft since CI passed on my fork

File: integration-tests/container-image/pom.xml (hunk 1 above)
Comment (geoand) [reply integration-tests/container-image/pom.xml]: I couldn't come up with anything better, so I went with your suggestion.

**Comment by iocanel** [comment]
> Looks good on the first sight. Just wondering if `container-image-s2i` also collides with `container-image-{docker,jib}`?

Yes, it does.

**Review by iocanel — APPROVED** [review]
LGTM

File: test-framework/junit5-internal/src/main/java/io/quarkus/test/QuarkusProdModeTest.java [hunk 2]
```diff
@@ -408,6 +454,32 @@ private void ensureApplicationStartupOrFailure() throws IOException {
...
+    public void interceptBeforeEachMethod(Invocation<Void> invocation, ReflectiveInvocationContext<Method> invocationContext,
+            ExtensionContext extensionContext) throws Throwable {
+        doIntercept(invocation);
+    }
+
+    @Override
+    public void interceptTestMethod(Invocation<Void> invocation, ReflectiveInvocationContext<Method> invocationContext,
+            ExtensionContext extensionContext) throws Throwable {
+        doIntercept(invocation);
+    }
```
Comment (Ladicek) [inline test-framework/junit5-internal/src/main/java/io/quarkus/test/QuarkusProdModeTest.java:473]: Do we need to also intercept `AfterAll` / `AfterEach`? And all the other methods :-)

**Review by Ladicek — APPROVED** [review]
LGTM, with one minor question.

File: test-framework/junit5-internal/src/main/java/io/quarkus/test/QuarkusProdModeTest.java (hunk 2 above)
Comment (geoand) [reply test-framework/junit5-internal/src/main/java/io/quarkus/test/QuarkusProdModeTest.java:473]: There might will be situations were we have to, but they are unlikely to ever come up with this kind of internal test (hopefully :laughing:). 
I say we leave it as is for now and fix if it ever comes up.

**Comment by geoand** [comment]
@rsvoboda OK with you as well?

**Comment by rsvoboda** [comment]
Looking now

**Review by rsvoboda — APPROVED** [review]
Approving

And the error message looks better now:
java.lang.IllegalStateException: quarkus-container-image-jib and quarkus-container-image-docker were detected, at most one container-image extension can be present.

**Comment by geoand** [comment]
Cool, thanks!

**Pull request merged**